lib/mozart_fetcher/http_client.ex

```diff
@@ -13,26 +13,36 @@ defmodule HTTPClient do
         hackney: [pool: :origin_pool]
       ]
 
-      case make_request(sanitise(endpoint), headers, options) do
-        {:error, %HTTPoison.Error{reason: "closed"}} ->
-          handle_response(make_request(sanitise(endpoint), headers, options))
-        {k, response} -> handle_response({k, response})
+      case log_errors_and_return(make_request(sanitise(endpoint), headers, options, client)) do
+        {:error, %HTTPoison.Error{reason: :closed}} ->
+          make_request(sanitise(endpoint), headers, options, client)
+
+        {:error, %HTTPoison.Error{reason: :timeout}} ->
```

Comment:
if upstream starts failing under the high load, do we want to retry?

Reply:
once yes

Reply:
I think it's irrelevant, assembler waits a bit longer than 3 seconds then gives up. If we timeout and retry assembler probably wont be listening.